auth/credentials/internal/externalaccount/x509_provider.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package externalaccount
+
+import (
+	"context"
+	"crypto/tls"
+	"net/http"
+	"time"
+
+	"cloud.google.com/go/auth/internal/transport/cert"
+)
+
+type x509Provider struct {
+	certificateConfigLocation string
+	cachedClient              *http.Client
+}
+
+func (xp *x509Provider) providerType() string {
+	return x509ProviderType
+}
+
+func (xp *x509Provider) subjectToken(ctx context.Context) (string, error) {
+	return "", nil
```

Comment:
I wonder if this is the wrong abstraction. If there is no subject token this is not really a provider of tokens.

Reply:
So rename this to emptyProvider or something like that and then have a seperate type to handle the client creation?

Reply:
It feels like it should just not impl this interface because it is not a provider of token.

Reply:
It seems like exchanging subject token with access token is the core focus of the "externalaccount" credentials type, and since x509 is not doing that, it should sit outside of "externalaccount" entirely, and be it's own credentials type? I'm not familiar with the original externalaccount effort so this is just based on what I can observe from this PR.

Reply:
This should definitely live in externalaccount, the point of externalaccount is not necessarily to just exchange subject tokens, its to exchange some 3rd party credential for a GCP credential. In the past it has always been subject tokens, but with the new X509 credential type, a certificate is being used. This also maps to the external_account type in the ADC configuration, which the new X509 still falls under.

If we don't have a provider here that implements the interface, would you prefer just returning an anonymous function in NewTokenProvider that returns an empty string? @codyoss I don't have strong feelings either way, we just need something that implements subjectTokenProvider to use when createing a new tokenProvider.

Reply:
Gotcha. I think it may be more clear to introduce a concept of an "emptySubjectTokenProvider" to use along with X509 logic. I understand the existing code is structured to always expect a "subject token", but at the end of the day we really only care about implementing the auth.TokenProvider interface. I'll let you and Cody make the call.